addons/logistic_x/travel.py

```diff
@@ -218,17 +217,34 @@ def get_pricelist(self, cr, uid, partner, context=None):
                 raise osv.except_osv(_('Error!'), _('Order cannot be created because not sale pricelist exists!'))
         return self.pool['product.pricelist'].browse(cr, uid, pricelist_id, context=context)
 
+
     def invoice_line_create(self, cr, uid, ids, context=None):
         if context is None:
             context = {}
-
         create_ids = []
-        for travel in self.browse(cr, uid, ids, context=context):
-            vals = self._prepare_order_line_invoice_line(cr, uid, travel, context=context)
-            if vals:
-                inv_line_id = self.pool.get('account.invoice.line').create(cr, uid, vals, context=context)
-                self.write(cr, uid, [travel.id], {'invoice_line_id': inv_line_id}, context=context)
-                create_ids.append(inv_line_id)
+        product_currency = {}
+        for o in self.browse(cr, uid, ids, context=context):
```

Comment:
Entiendo que con esto lo que se quiere es tener un diccionario donde se pisan las claves si un producto aparece más de una vez. Pero veo que lo de la "currency" después no se usa en ningun lado, o sea, es un poco vueltero y mejorable. 
Lo podemos mejorar haciendo algo así...
Supongamos que yo puedo obtener un listado de productos, eso podría ser algo como "product_ids = [x.product_id.id for x in travels if x.product_id]
En product_ids tenemos una lista tipo [0 , 1, 2, 1, 3 , 1]
si a esa lisat la transformamos en un conjunto (http://docs.python.org.ar/tutorial/3/datastructures.html#conjuntos), automaticamente los elementos repetidos se eliminan, eso sería "set(product_ids)"

Pero luego queremos una lista, entonces podes hacer:

for product_id in list(set(product_ids)):

Podes probarlo? Esta bien así... pero es para que de paso veas como usar conjuntos y demás...